### ISSUE TITLE
[MEAR-214] RarModule not seen as standard artifact type

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-ear-plugin#390]
**[Luke Hamaty](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=wlhamaty)** opened **[MEAR-214](https://issues.apache.org/jira/browse/MEAR-214?redirect=false)** and commented

Configuration with a RarModule e.g.:

```xml
				<plugin>
					<groupId>org.apache.maven.plugins</groupId>
					<artifactId>maven-ear-plugin</artifactId>
					<version>2.10</version>
					<configuration>
						<defaultLibBundleDir>APP-INF/lib</defaultLibBundleDir>
						<skinnyWars>true</skinnyWars>
						<skipClassPathModification>true</skipClassPathModification>
<generateApplicationXml>true</generateApplicationXml>
						<modules>
							<!-- Resource Adapter for SSA Application -->
							<rarModule>
								<groupId>some-group</groupId>
								<artifactId>some-rar</artifactId>
								<bundleFileName>some.rar</bundleFileName>
							</rarModule>
```

Results in an IllegalStateException:

```
java.lang.IllegalStateException: Artifact type[connector] is not a standard Ear artifact type[[jar, ejb, ejb3, par, ejb-client, app-client, rar, war, sar, wsr, har]]
	at org.apache.maven.plugin.ear.util.ArtifactTypeMappingService.isMappedToType(ArtifactTypeMappingService.java:127)
	at org.apache.maven.plugin.ear.util.ArtifactRepository.getArtifacts(ArtifactRepository.java:138)
	at org.apache.maven.plugin.ear.util.ArtifactRepository.getUniqueArtifact(ArtifactRepository.java:75)
	at org.apache.maven.plugin.ear.AbstractEarModule.resolveArtifact(AbstractEarModule.java:144)
	at org.apache.maven.plugin.ear.AbstractEarMojo.execute(AbstractEarMojo.java:228)
	at org.apache.maven.plugin.ear.GenerateApplicationXmlMojo.execute(GenerateApplicationXmlMojo.java:162)
	at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:490)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:694)

```

This does not occur with 2.9.1.
This appears to have been introduced with revision 1645331 of RarModule.java, which incorrectly changed type from "rar" to RAR_MODULE

---

**Affects:** 2.10

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Fixed in [r1660473](http://svn.apache.org/r1660473).
Thanks for finding it.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Hi Luke, are you able to test a SNAPSHOT version just to be sure i fixed what the problem was?

**Comment by jira-importer** [comment]
**[Luke Hamaty](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=wlhamaty)** commented

This works correctly:
maven-ear-plugin: resolved to version 2.11-20150218.152705-91

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Cool thanks for the feedback.
